tools/run_tests/run_tests.py

```diff
@@ -1570,7 +1570,12 @@ def _build_and_run(
     for antagonist in antagonists:
       antagonist.kill()
     if args.bq_result_table and resultset:
-      upload_results_to_bq(resultset, args.bq_result_table, args, platform_string())
+      # There are transient errors with uploading to BQ from Kokoro and they
+      # should not cause the build to fail.
+      try:
+        upload_results_to_bq(resultset, args.bq_result_table, args, platform_string())
```

Comment:
I'm not sure this is a good idea.  If failure to upload is silent, we will not be aware of any problems with uploading the test results if it suddenly stops working (and we will have limited flakiness data from master and limited data from PRs).
This is even a bit worse, because for successful test suite, run_tests_matrix.py eats all the output, so there will never be any useful output in the logs to check if the results were uploaded properly.
Can we 1. try to figure out why results fail to be uploaded (it says deadline exceeded, so perhaps we can just increase the timeout for the upload)  2. retry the upload

Reply:
https://github.com/grpc/grpc/pull/12839